scheduler.rst

```diff
@@ -0,0 +1,328 @@
+
+If you go back to your scenario regarding reports generation based on your customer preferences.
+If the basic frequency is set to a daily basis, you will need to implement a custom trigger due to the specific requirement of not generating reports during public holiday periods::
+
+    // src/Scheduler/Trigger/NewUserWelcomeEmailHandler.php
+    namespace App\Scheduler\Trigger;
+
+    class ExcludeHolidaysTrigger implements TriggerInterface
+    {
+        public function __construct(private TriggerInterface $inner)
+        {
+        }
+
+        public function __toString(): string
+        {
+            return $this->inner.' (except holidays)';
+        }
+
+        public function getNextRunDate(\DateTimeImmutable $run): ?\DateTimeImmutable
+        {
+            if (!$nextRun = $this->inner->getNextRunDate($run)) {
+                return null;
+            }
+
+            while (!$this->isHoliday($nextRun) { // loop until you get the next run date that is not a holiday
+                $nextRun = $this->inner->getNextRunDate($nextRun);
+            }
+
+            return $nextRun;
+        }
+
+        private function isHoliday(\DateTimeImmutable $timestamp): bool
+        {
+            // app specific logic to determine if $timestamp is on a holiday
+            // returns true if holiday, false otherwise
+        }
+    }
+
+Then, you would have to define your RecurringMessage
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        RecurringMessage::trigger(
+            new ExcludeHolidaysTrigger( // your custom trigger wrapper
+                CronExpressionTrigger::fromSpec('@daily'),
+            ),
+            new SendDailySalesReports(),
+        );
+
+The RecurringMessages must be attached to a Schedule::
+
+    // src/Scheduler/MyScheduleProvider.php
+    namespace App\Scheduler;
+
+    #[AsSchedule('uptoyou')]
+    class MyScheduleProvider implements ScheduleProviderInterface
+    {
+        public function getSchedule(): Schedule
+        {
+            return $this->schedule ??= (new Schedule())
+                ->with(
+                    RecurringMessage::trigger(
+                        new ExcludeHolidaysTrigger( // your custom trigger wrapper
+                            CronExpressionTrigger::fromSpec('@daily'),
+                        ),
+                    new SendDailySalesReports()),
+                    RecurringMessage::cron(‘3 8 * * 1’, new CleanUpOldSalesReport())
+
+                );
+        }
+    }
+
+So, this RecurringMessage will encompass both the trigger, defining the generation frequency of the message, and the message itself, the one to be processed by a specific handler.
+
+Efficient management with Symfony Scheduler
+-------------------------------------------
+
+However, if your worker becomes idle, since the messages from your schedule are generated on-the-fly by the schedulerTransport,
```

Comment:
We haven't talked about workers or `messenger:consume` yet. I think we should at least show them. The story I want to see is:

A) Creating a message & handler of some sort (this can be very quick and point to Messenger, and the handler can be empty, but we should show it)
B) Creating a schedule and setting up this message in a recurring way
C) Running `messenger:consume` - and talking a bit about how that works and what to expect.